gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_page.dart

```diff
@@ -26,83 +29,82 @@ class SubscribeNowPage extends StatelessWidget {
+                    // Better check if it's still mounted in the widget tree.
+                    if (!context.mounted) return;
 
-                        subs.fold(
-                          ifLeft: (status) {
-                            ScaffoldMessenger.of(context).showSnackBar(
-                              SnackBar(
-                                width: 200.0,
-                                behavior: SnackBarBehavior.floating,
-                                content: Center(
-                                  child: Padding(
-                                    padding: const EdgeInsets.symmetric(
-                                      vertical: 2.0,
-                                      horizontal: 16.0,
-                                    ),
-                                    child: Text(status.localize(lang)),
-                                  ),
+                    subs.fold(
+                      ifLeft: (status) {
+                        ScaffoldMessenger.of(context).showSnackBar(
+                          SnackBar(
+                            width: 200.0,
+                            behavior: SnackBarBehavior.floating,
+                            content: Center(
+                              child: Padding(
+                                padding: const EdgeInsets.symmetric(
+                                  vertical: 2.0,
+                                  horizontal: 16.0,
                                 ),
+                                child: Text(status.localize(lang)),
                               ),
-                            );
-                          },
-                          ifRight: onSubscriptionUpdate,
+                            ),
+                          ),
                         );
-                      }
-                    : null,
-                child: Text(lang.getUbuntuPro),
-              ),
-              const SizedBox(width: 8.0),
-            ],
-          ],
-        ),
-        const Padding(
-          padding: EdgeInsets.only(top: 16.0, bottom: 24.0),
-          child: Divider(thickness: 0.2),
-        ),
+                      },
+                      ifRight: onSubscriptionUpdate,
+                    );
+                  }
+                : null,
+            child: Text(lang.getUbuntuPro),
+          ),
+        ],
+      ],
+      right: [
         ProTokenInputField(
-          onApply: (token) {
-            model.applyProToken(token).then(onSubscriptionUpdate);
-          },
+          onSubmit: model.canSubmit ? () => trySubmit(model) : null,
+          controller: controller,
         ),
       ],
+      navigationRow: NavigationRow(
+        showBack: false,
+        onBack: null,
+        onNext: model.canSubmit ? () => trySubmit(model) : null,
+        nextText: lang.attach,
+      ),
     );
   }
 
+  void trySubmit(SubscribeNowModel model) {
+    model.applyProToken(model.token.token!).then(onSubscriptionUpdate);
+    model.token.clear();
```

Comment:
This could be `model.clear()` if you implement the method in the model per my suggestion above.